test/fixtures/smartypants.md

```diff
@@ -0,0 +1,3 @@
+# Hello "World"
+
+This is -- excuse my French -- awesome!
```

Comment:
@trieloff Isn't this supposed to be a mdash instead?

I think the rule is:
- use simple hyphen for composed nouns: `t-shirt`
- use ndash for number/date ranges: `2018–2020`
- use mdash for inlined parentheses: `This is — excuse my French — awesome!`